---
title: Differential gene expression upon lncRNA knockdown with siRNA, LNA and CRISPRi
author: Aaron Lun
date: 12 September 2016
output:
  html_document:
    toc: true
    toc_float: true
    depth: 3
    number_sections: true
    theme: united 
    highlight: tango    
    fig_caption: false
---

```{r, echo=FALSE, results='hide'}
knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
knitr::opts_knit$set(width=100)
options(width=100)
```

# Introduction

## Primary data loading

Loading in the counts, and ignoring the first column of gene lengths.

```{r}
all.counts <- read.table("../../data/genic_counts.tsv", header=TRUE, row.names=1, sep="\t",
        colClasses=c(list("character", "integer"), as.list(rep("integer", 768))))
lengths <- all.counts[,1]
all.counts <- all.counts[,-1]
dim(all.counts)
```

Summing over the technical replicates.

```{r}
library(edgeR)
lib.name <- sub("_.*", "", colnames(all.counts))
all.counts <- sumTechReps(all.counts, lib.name)
dim(all.counts)
```

Loading in the experimental design, and checking that all the samples are present.

```{r}
metadata <- read.table("../../data/metadata.tsv", stringsAsFactors=FALSE, header=TRUE, sep="\t")
m <- match(colnames(all.counts), metadata$Library)
stopifnot(all(!is.na(m)))
metadata <- metadata[m,]
```

Removing the unnecessary H19 samples.

```{r}
is.h19 <- grepl("H19", metadata$Condition)
metadata <- metadata[!is.h19,]
all.counts <- all.counts[,!is.h19]
```

Generating a `DGEList` object.

```{r}
require(edgeR)
y <- DGEList(all.counts)
y$samples$group <- metadata$Condition
table(y$samples$group)
```

# Data exploration

Exploring the libraries.

```{r}
colors <- rainbow(length(unique(y$samples$group)))
adjc <- cpm(y, log=TRUE, prior=3)
mds.out <- plotMDS(adjc, col=colors[as.integer(factor(y$samples$group))], pch=16) 
```

Colouring by type of experiment (CRISPRi, siRNA or LNA).

```{r}
col <- rep("grey80", ncol(y)) # cells alone
col[metadata$LOF=="CRISPRi"] <- "orange"
col[metadata$LOF=="LNA"] <- "blue" 
col[metadata$LOF=="RNA interference"] <- "red" 
plotMDS(mds.out, col=col, pch=16, label=1:ncol(y))
```

Colouring by batch.

```{r}
col <- c("purple", "pink", "black", "grey80", "dodgerblue")[metadata$Batch]
plotMDS(mds.out, col=col, pch=16)
```

Checking that the samples express the genes they're meant to express (or not).

```{r}
g <- factor(metadata$Condition)
par(mar=c(12.1,4.1,4.1,2.1))
plot(as.integer(g), adjc["ENSG00000234771",], xaxt="n", ylab="log-CPM", main="289", xlab="")
axis(1, at=seq_len(nlevels(g)), levels(g), las=2)
par(mar=c(12.1,4.1,4.1,2.1))
plot(as.integer(g), adjc["Cas9",], xaxt="n", ylab="log-CPM", main="Cas9", xlab="")
axis(1, at=seq_len(nlevels(g)), levels(g), las=2)
```

We save these to file for prettier plotting.

```{r}
to.save <- data.frame(Group=g, t(adjc[c("ENSG00000234771", "Cas9"),]))
to.save <- to.save[order(to.save$Group),]
write.table(file="key_exprs.tsv", to.save, sep="\t", quote=FALSE, col.names=NA)
```

# Filtering, normalization and annotation

Removing lowly-expressed genes (mean of 3 across all libraries).
Also removing Cas9, as it's not an interesting biological gene.

```{r}
ab <- aveLogCPM(y)
keep <- ab > aveLogCPM(3, mean(y$samples$lib.size)) & rownames(y)!="Cas9"
y <- y[keep,]
sum(keep)
```

Normalizing to remove composition biases.

```{r}
y <- calcNormFactors(y)
hist(y$samples$norm.factors, col="grey80", xlab="Normalization factors")
```

Adding annotation.

```{r, message=FALSE}
require(org.Hs.eg.db)
symb <- mapIds(org.Hs.eg.db, keys=rownames(y), keytype="ENSEMBL", column="SYMBOL")
y$genes <- data.frame(Symbol=symb, row.names=rownames(y))
```

# Modelling the mean-variance relationship

Setting up a design matrix.
We use a one-way layout where each group gets its own coefficient in the matrix.
We also add blocking factors for the various experimental terms, including the condition and the batch.
(The first factor for each batch needs to be removed to achieve full rank. 
This is except for the first batch, where the blocking factor is automatically removed by `model.matrix`.)

```{r}
block <- paste0(metadata$Batch, metadata$Experiment)
g <- factor(metadata$Condition)
design <- model.matrix(~0 + g + block)
design <- design[,! colnames(design) %in% c("block21", "block36", "block41", "block52")]
dim(design)
```

Modelling the mean-variance trend with `voom`.
Also computing library-specific weights to downweight any outlier libraries within each group.

```{r, fig.width=10, fig.height=6}
v.all <- voomWithQualityWeights(y, design, plot=TRUE)
```

Fitting the linear model with robust empirical Bayes shrinkage to stabilize the variance estimates.

```{r}
fit <- lmFit(v.all, design)
fit <- eBayes(fit, robust=TRUE)
plotSA(fit)
```

Also checking the negative binomial dispersion estimates from `edgeR`.

```{r}
y <- estimateDisp(y, design)
summary(y$trended)
plotBCV(y)
```

# Testing for differential expression

## Setting up output folders

Setting up some output folders, and some functions to store results.

```{r}
dir.create("results_de", showWarning=FALSE)
dir.create("results_lfc", showWarning=FALSE)
WRITE <- function(fit, fname, lfc=FALSE) {
    fname <- paste0(fname, ".txt")
    if (lfc) { 
        res <- topTreat(fit, n=Inf, sort.by="p")
        fname <- file.path("results_lfc", fname)
    } else {
        res <- topTable(fit, n=Inf, sort.by="p")
        fname <- file.path("results_de", fname)
    }
    write.table(file=fname, res, row.names=FALSE, quote=FALSE, sep="\t")
    head(res)
}
```

```{r progenitor, echo=FALSE, eval=FALSE}
fit2 <- contrasts.fit(fit, con)
fit2 <- treat(fit2, lfc=0.5, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname, lfc=TRUE)
```

## Contrasts for siRNA

### Effect of siRNA transfection plus any sequence-dependent off-target effects 

Testing for DE between Ambion control siRNA and cells.

```{r}
fname <- "siRNA_Ambion_vs_cells"
con <- makeContrasts(
    gRNA_interference.wild_type_genotype.Ambion_control.batch_1 
        - gnone.wild_type_genotype.none.batch_1,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between Dharmacon control siRNA and cells.

```{r}
fname <- "siRNA_Dharmacon_vs_cells"
con <- makeContrasts(
    gRNA_interference.wild_type_genotype.Dharmacon_control.batch_1 
         - gnone.wild_type_genotype.none.batch_1,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Minimal sequence-dependent off-target effects

Testing for DE between siRNA controls.

```{r}
fname <- "siRNA_Ambion_vs_Dharmacon"
con <- makeContrasts(
    gRNA_interference.wild_type_genotype.Ambion_control.batch_1 
        - gRNA_interference.wild_type_genotype.Dharmacon_control.batch_1, 
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Effect of actual knockdown plus any off-target effects

Testing for DE between 289 siRNA and the Ambion control.

```{r}
fname <- "siRNA_289_vs_Ambion"
con <- makeContrasts(
    gRNA_interference.wild_type_genotype.289_siRNA.batch_1 
        - gRNA_interference.wild_type_genotype.Ambion_control.batch_1,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between 289 siRNA and the Dharmacon control.

```{r}
fname <- "siRNA_289_vs_Dharmacon"
con <- makeContrasts(
    gRNA_interference.wild_type_genotype.289_siRNA.batch_1 
        - gRNA_interference.wild_type_genotype.Dharmacon_control.batch_1,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

## Contrasts for LNA

### Effect of transfection with MAX reagent

Testing for DE between transfected cels and non-transfected cells.

```{r}
fname <- "LNA_trans_vs_cells"
con <- makeContrasts(
    gLNA.wild_type_genotype.Transfection_control.batch_4 
        - gnone.wild_type_genotype.none.batch_4,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Effect of LNA transfection plus off-target effects 

Testing for DE between LNA control A and transfected cells.

```{r}
fname <- "LNA_controlA_vs_trans"
con <- makeContrasts(
    gLNA.wild_type_genotype.Negative_control_A.batch_4 
        - gLNA.wild_type_genotype.Transfection_control.batch_4,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between LNA control B and transfected cells.

```{r}
fname <- "LNA_controlB_vs_trans"
con <- makeContrasts(
    gLNA.wild_type_genotype.Negative_control_B.batch_4 
        - gLNA.wild_type_genotype.Transfection_control.batch_4,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Minimal effect of sequence-dependent off-target effects

Testing for DE between LNA controls A and B.

```{r}
fname <- "LNA_controlA_vs_controlB"
con <- makeContrasts(
    gLNA.wild_type_genotype.Negative_control_A.batch_4 
        - gLNA.wild_type_genotype.Negative_control_B.batch_4, 
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Effect of actual knockdown plus any off-target effects

Testing for DE between 289 LNA2 knockdown and control B.

```{r}
fname <- "LNA_289.2_vs_controlB"
con <- makeContrasts(
    gLNA.wild_type_genotype.289_LNA.batch_1 
        - gLNA.wild_type_genotype.Negative_control_B.batch_1,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between 289 LNA2 knockdown and control A, via the shared control B's (differential response, effectively).

```{r}
fname <- "LNA_289.2_vs_controlA"
con <- makeContrasts(
    (gLNA.wild_type_genotype.289_LNA.batch_1 
        - gLNA.wild_type_genotype.Negative_control_B.batch_1) - 
        (gLNA.wild_type_genotype.Negative_control_A.batch_4 
        - gLNA.wild_type_genotype.Negative_control_B.batch_4),
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

## Contrasts for CRISPRi (clonal)

### Effect of dCas9-KRAB expression and cloning

Testing for DE between dCas9-expressing clone 2 and cells (first batch).

```{r}
fname <- "CRISPRi_clone2_vs_cells_I"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.none.batch_1 
        - gnone.wild_type_genotype.none.batch_1,
    levels=design)
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between dCas9-expressing clone 2 and cells (second batch).

```{r}
fname <- "CRISPRi_clone2_vs_cells_II"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.none.batch_3 
        - gnone.wild_type_genotype.none.batch_3,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between dCas9-expressing clone 1 and cells.

```{r}
fname <- "CRISPRi_clone1_vs_cells"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_1.none.batch_3 
        - gnone.wild_type_genotype.none.batch_3,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between dCas9-expressing clone 3 and cells.

```{r}
fname <- "CRISPRi_clone3_vs_cells"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_3.none.batch_3 
        - gnone.wild_type_genotype.none.batch_3,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Minimal founder effects of cloning 

Testing for DE between clones 1 and 2.

```{r}
fname <- "CRISPRi_clone1_vs_clone2"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_1.none.batch_3 
        - gCRISPRi.dCas9_KRAB_clone_2.none.batch_3,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between clones 1 and 3.

```{r}
fname <- "CRISPRi_clone1_vs_clone3"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_1.none.batch_3 
        - gCRISPRi.dCas9_KRAB_clone_3.none.batch_3,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between clones 2 and 3.

```{r}
fname <- "CRISPRi_clone2_vs_clone3"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.none.batch_3 
        - gCRISPRi.dCas9_KRAB_clone_3.none.batch_3,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Effect of guide transfection plus off-target effects

Testing for DE between the negative guide 2 and clone 2 (first batch).

```{r}
fname <- "CRISPRi_negguide2_vs_clone2_I"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_2.batch_1 
        - gCRISPRi.dCas9_KRAB_clone_2.none.batch_1,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between the negative guide 2 and clone 2 (second batch).

```{r}
fname <- "CRISPRi_negguide2_vs_clone2_II"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_2.batch_2 
        - gCRISPRi.dCas9_KRAB_clone_2.none.batch_2,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between the negative guide 1 and clone 2.

```{r}
fname <- "CRISPRi_negguide1_vs_clone2"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_1.batch_2 
        - gCRISPRi.dCas9_KRAB_clone_2.none.batch_2,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Minimal effect of sequence-dependent off-target effects

Testing for DE between guides 1 and 2.

```{r}
fname <- "CRISPRi_negguide1_vs_negguide2"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_1.batch_2 
        - gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_2.batch_2,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Effect of actual 289 knockdown plus any off-target effects 

Testing for DE between 289 guide 1 and negative guide 2.

```{r}
fname <- "CRISPRi_289.1_vs_negguide2"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.289_guide_1.batch_1 
        - gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_2.batch_1,   
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between 289 guide 9 and negative guide 2.

```{r}
fname <- "CRISPRi_289.9_vs_negguide2"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.289_guide_9.batch_1 
        - gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_2.batch_1,   
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

<!--

### Effect of actual H19 knockdown plus any off-target effects 


Testing for DE between H19 guide 1 and negative guide 1.

```{r, eval=FALSE}
fname <- "CRISPRi_H19.1_vs_negguide1"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.H19_guide_1.batch_2 - 
        gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_1.batch_2,
    levels=design)
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor", eval=FALSE}
```

Testing for DE between H19 guide 1 and negative guide 2.

```{r, eval=FALSE}
fname <- "CRISPRi_H19.1_vs_negguide2"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.H19_guide_1.batch_2 - 
        gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_2.batch_2,
    levels=design)        
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, eval=FALSE, ref.label="progenitor", eval=FALSE}
```

Testing for DE between H19 guide 2 and negative guide 1.

```{r, eval=FALSE}
fname <- "CRISPRi_H19.2_vs_negguide1"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.H19_guide_2.batch_2 - 
        gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_1.batch_2,
    levels=design)        
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, eval=FALSE, ref.label="progenitor"}
```

Testing for DE between H19 guide 2 and negative guide 2.

```{r, eval=FALSE}
fname <- "CRISPRi_H19.2_vs_negguide2"
con <- makeContrasts(
    gCRISPRi.dCas9_KRAB_clone_2.H19_guide_2.batch_2 - 
        gCRISPRi.dCas9_KRAB_clone_2.Negative_control_guide_2.batch_2,
    levels=design)        
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, eval=FALSE, ref.label="progenitor"}
```
-->

## Contrasts for CRISPRi (heterogeneous)

### Testing for the effect of dCas9 in heterogeneous populations

Comparing HeLa cells with those that have been transduced and selected for dCas9-KRAB expression in a heterogeneous population.

```{r}
fname <- "CRISPRi_het_BFP_vs_cells"
con <- makeContrasts(
    gCRISPRi.heterogenous_dCas9_KRAB.none.batch_5 
        - gnone.wild_type_genotype.none.batch_5,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Testing for the effect of control guide transfection

Testing for DE after transfection of negative control to heterogeneous populations.

```{r}
fname <- "CRISPRi_het_Nc.1_vs_BFP"
con <- makeContrasts(
    gCRISPRi.heterogenous_dCas9_KRAB.Negative_control_guide_1.batch_5 
        - gCRISPRi.heterogenous_dCas9_KRAB.none.batch_5,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Now for the other guide.

```{r}
fname <- "CRISPRi_het_Nc.2_vs_BFP"
con <- makeContrasts(
    gCRISPRi.heterogenous_dCas9_KRAB.Negative_control_guide_2.batch_5 
        - gCRISPRi.heterogenous_dCas9_KRAB.none.batch_5,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Testing for DE between the negative guides.

```{r}
fname <- "CRISPRi_het_Nc.1_vs_Nc.2"
con <- makeContrasts(
    gCRISPRi.heterogenous_dCas9_KRAB.Negative_control_guide_1.batch_5 
       - gCRISPRi.heterogenous_dCas9_KRAB.Negative_control_guide_2.batch_5,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### Testing for the effect of 289 knockdown

Comparing to the first negative guide.

```{r}
fname <- "CRISPRi_het_289.9_vs_Nc.1"
con <- makeContrasts(
    gCRISPRi.heterogenous_dCas9_KRAB.289_guide_9.batch_5 
        - gCRISPRi.heterogenous_dCas9_KRAB.Negative_control_guide_1.batch_5,
    levels=design)
colnames(con) <- fname
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

Comparing to the second negative guide.

```{r}
fname <- "CRISPRi_het_289.9_vs_Nc.2"
con <- makeContrasts(
    gCRISPRi.heterogenous_dCas9_KRAB.289_guide_9.batch_5 
        - gCRISPRi.heterogenous_dCas9_KRAB.Negative_control_guide_2.batch_5,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

<!--
### Testing for the effect of H19 knockdown

Comparing to the first negative guide.

```{r, eval=FALSE}
fname <- "CRISPRi_het_H19_vs_Nc.1"
con <- makeContrasts(
    gCRISPRi.heterogenous_dCas9_KRAB.H19_guide_2.batch_5  
        - gCRISPRi.heterogenous_dCas9_KRAB.Negative_control_guide_1.batch_5,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor", eval=FALSE}
```

Comparing to the second negative guide.

```{r, eval=FALSE}
fname <- "CRISPRi_het_H19_vs_Nc.2"
con <- makeContrasts(
    gCRISPRi.heterogenous_dCas9_KRAB.H19_guide_2.batch_5 
         - gCRISPRi.heterogenous_dCas9_KRAB.Negative_control_guide_2.batch_5,
    levels=design)
colnames(con) <- fname   
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor", eval=FALSE}
```
-->

# Session information

```{r}
saveRDS(file="object.rds", y)
sessionInfo()
```
