---
title: Differential gene expression upon positive control knockdowns
author: Aaron Lun
date: 5 March 2018
output:
  html_document:
    toc: true
    toc_float: true
    depth: 3
    number_sections: true
    theme: united
    highlight: tango
    fig_caption: false
---

```{r, echo=FALSE, results='hide'}
knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
knitr::opts_knit$set(width=100)
options(width=100)
if (!exists(".use.local")) { .use.local <- TRUE }
```

# Introduction

## Primary data loading

Loading in the counts, and ignoring the first column of gene lengths.

```{r, eval=.use.local, echo=.use.local}
all.counts <- read.table("../data/subcounts.tsv", header=TRUE, row.names=1, sep="\t",
    colClasses=c(list("character", "integer"), as.list(rep("integer", 51))))
lengths <- all.counts[,1]
all.counts <- all.counts[,-1]
dim(all.counts)
```

Summing over the technical replicates.

```{r}
library(edgeR)
lib.name <- sub(".*(D[0-9]+_D[0-9]+).*", "\\1", colnames(all.counts))
table(lib.name)
all.counts <- sumTechReps(all.counts, lib.name)
dim(all.counts)
```

Loading in the experimental design, and checking that all the samples are present.

```{r, eval=.use.local, echo=.use.local}
metadata <- read.table("../data/metadata.tsv", stringsAsFactors=FALSE, header=TRUE, sep="\t")
m <- match(colnames(all.counts), metadata$Library)
stopifnot(all(!is.na(m)))
metadata <- metadata[m,]
```

Generating a `DGEList` object.

```{r}
require(edgeR)
y <- DGEList(all.counts)
y$samples$group <- metadata$Condition
y$samples$block <- paste0(metadata$Batch, metadata$Experiment)
table(y$samples$group)
table(y$samples$block)
```

# Data exploration

Exploring the libraries.

```{r}
colors <- rainbow(length(unique(y$samples$group)))
adjc <- cpm(y, log=TRUE, prior=3)
mds.out <- plotMDS(adjc, col=colors[as.integer(factor(y$samples$group))], pch=16)
```

Checking that the samples express the genes they're meant to express (or not).

```{r}
g <- factor(metadata$Condition)

par(mar=c(12.1,4.1,4.1,2.1)) # chTOG
plot(as.integer(g), adjc["ENSG00000175216",], xaxt="n", ylab="log-CPM", main="chTOG", xlab="")
axis(1, at=seq_len(nlevels(g)), levels(g), las=2)

par(mar=c(12.1,4.1,4.1,2.1))
plot(as.integer(g), adjc["Cas9",], xaxt="n", ylab="log-CPM", main="Cas9", xlab="")
axis(1, at=seq_len(nlevels(g)), levels(g), las=2)

par(mar=c(12.1,4.1,4.1,2.1)) # MALAT1
plot(as.integer(g), adjc["ENSG00000251562",], xaxt="n", ylab="log-CPM", main="MALAT1", xlab="")
axis(1, at=seq_len(nlevels(g)), levels(g), las=2)
```

We save these to file for prettier plotting.

```{r}
to.save <- data.frame(Group=g, t(adjc[c("ENSG00000175216", "Cas9", "ENSG00000251562"),]))
to.save <- to.save[order(to.save$Group),]
write.table(file="key_exprs.tsv", to.save, sep="\t", quote=FALSE, col.names=NA)
```

# Filtering, normalization and annotation

Removing lowly-expressed genes (mean of 3 across all libraries).
Also removing Cas9, as it's not an interesting biological gene.

```{r}
ab <- aveLogCPM(y)
keep <- ab > aveLogCPM(3, mean(y$samples$lib.size)) & rownames(y)!="Cas9"
y <- y[keep,]
sum(keep)
```

Normalizing to remove composition biases.

```{r}
y <- calcNormFactors(y)
hist(y$samples$norm.factors, col="grey80", xlab="Normalization factors")
```

Adding annotation.

```{r, message=FALSE}
require(org.Hs.eg.db)
symb <- mapIds(org.Hs.eg.db, keys=rownames(y), keytype="ENSEMBL", column="SYMBOL")
y$genes <- data.frame(Symbol=symb, row.names=rownames(y))
```

# Modelling the mean-variance relationship

We use an additive model where each group gets its own coefficient in the matrix.
We also add blocking factors for the various experimental terms, including the condition and the batch.

```{r}
g <- factor(y$samples$group)
block <- factor(y$samples$block)
design <- model.matrix(~0 + g + block)
dim(design)
```

Converting the matrix to full rank.

```{r}
c(ncol(design), qr(design)$rank)
design <- design[,!colnames(design) %in% c("block64", "block67")]
c(ncol(design), qr(design)$rank)
```

Modelling the mean-variance trend with `voom`.
Also computing library-specific weights to downweight any outlier libraries within each group.

```{r, fig.width=10, fig.height=6}
v.all <- voomWithQualityWeights(y, design, plot=TRUE)
```

Fitting the linear model with robust empirical Bayes shrinkage to stabilize the variance estimates.

```{r}
fit <- lmFit(v.all, design)
fit <- eBayes(fit, robust=TRUE)
plotSA(fit)
```

Also checking the negative binomial dispersion estimates from `edgeR`.

```{r}
y <- estimateDisp(y, design)
summary(y$trended)
plotBCV(y)
```

# Testing for differential expression

## Setting up output folders

Setting up some output folders, and some functions to store results.

```{r}
dir.create("results_de", showWarning=FALSE)
dir.create("results_lfc", showWarning=FALSE)
WRITE <- function(fit, fname, lfc=FALSE) {
    fname <- paste0(fname, ".txt")
    if (lfc) {
        res <- topTreat(fit, n=Inf, sort.by="p")
        fname <- file.path("results_lfc", fname)
    } else {
        res <- topTable(fit, n=Inf, sort.by="p")
        fname <- file.path("results_de", fname)
    }
    write.table(file=fname, res, col.names=NA, quote=FALSE, sep="\t")
    head(res)
}
```

```{r progenitor, echo=FALSE, eval=FALSE}
fit2 <- contrasts.fit(fit, con)
fit2 <- treat(fit2, lfc=0.5, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname, lfc=TRUE)
```

## Contrasts for siRNA

Testing for DE between TOG-knockdowns and the Dharmacon control.

```{r}
fname <- "siRNA_TOG_vs_Dharmacon"
con <- makeContrasts(levels=design,
    gRNA_interference.wild_type_genotype.Ch_TOG_siRNA.batch_6 
    - gRNA_interference.wild_type_genotype.Dharmacon_control.batch_6)
colnames(con) <- fname
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

## Contrasts for LNA 

Testing for DE between MALAT1-knockdowns and the Dharmacon control.

```{r}
fname <- "LNA_MALAT1_vs_controlA"
con <- makeContrasts(levels=design,
    gLNA.wild_type_genotype.MALAT1_LNA.batch_6
    - gLNA.wild_type_genotype.Negative_control_A.batch_6)
colnames(con) <- fname
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

## Contrasts for CRISPRi

### TOG

Testing for DE between TOG-knockdowns and the control guide.

```{r}
fname <- "CRISPRi_het_TOG_vs_negguide2"
con <- makeContrasts(levels=design,
    gCRISPRi.heterogeneous_dCas9_KRAB.Ch_TOG_guide_1.batch_6
    - gCRISPRi.heterogeneous_dCas9_KRAB.Negative_control_guide_2.batch_6)
colnames(con) <- fname
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

### MALAT1 

Testing for DE between MALAT1-knockdowns and the control guide.

```{r}
fname <- "CRISPRi_het_MALAT1_vs_negguide2"
con <- makeContrasts(levels=design,
    gCRISPRi.heterogeneous_dCas9_KRAB.MALAT1_guide_1.batch_6
    - gCRISPRi.heterogeneous_dCas9_KRAB.Negative_control_guide_2.batch_6)
colnames(con) <- fname
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, ref.label="progenitor"}
```

<!--
## Contrasts for the drugs

After treatment with Monastrol.

```{r, eval=FALSE}
fname <- "Monastrol"
con <- makeContrasts(gMonastrol - ghela_cells, levels=design)
colnames(con) <- fname
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, eval=FALSE, ref.label="progenitor"}
```

After treatment with NOCO.

```{r, eval=FALSE}
fname <- "NOCO"
con <- makeContrasts(gNOCO - ghela_cells, levels=design)
colnames(con) <- fname
fit2 <- contrasts.fit(fit, con)
fit2 <- eBayes(fit2, robust=TRUE)
summary(decideTests(fit2))
WRITE(fit2, fname)
```

Repeating with a log-fold change threshold of 0.5.

```{r, eval=FALSE, ref.label="progenitor"}
```
-->

# Session information

```{r}
saveRDS(file="object.rds", y)
sessionInfo()
```
